src/controllers/capp.go

```diff
@@ -178,6 +189,71 @@ func (c *cappController) GetCappState(namespace, name string) (types.GetCappStat
 	return cappState, nil
 }
 
+func (c *cappController) GetCappDNS(namespace, name string) (types.GetDnsResponse, error) {
+	c.logger.Debug(fmt.Sprintf("Trying to fetch dns related to capp %q in namespace %q", name, namespace))
+
+	recordsK8S := &dnsrecordv1alpha1.CNAMERecordList{}
+
+	listOptions := prepareDNSListOptions(namespace, name)
+
+	err := c.client.List(c.ctx, recordsK8S, listOptions)
+	if err != nil {
+		c.logger.Error(fmt.Sprintf("Could not fetch dns related to capp %q in namespace %q with error: %v", name, namespace, err.Error()))
+		return types.GetDnsResponse{}, err
+	}
+
+	if len(recordsK8S.Items) == 0 {
+		_, err := c.GetCapp(namespace, name)
+		if err != nil {
+			return types.GetDnsResponse{}, err
+		}
+		return types.GetDnsResponse{}, nil
+	}
+
+	return types.GetDnsResponse{Records: parseDNSList(recordsK8S)}, nil
+}
+
+// parseDnsList converts dns from k8s object to the response dns type.
+func parseDNSList(recordsK8S *dnsrecordv1alpha1.CNAMERecordList) []types.DNS {
+	var records []types.DNS
+
+	for _, record := range recordsK8S.Items {
+		syncedStatus := corev1.ConditionFalse
+		readyStatus := corev1.ConditionFalse
+		dnsStatus := corev1.ConditionFalse
+
+		for _, condition := range record.Status.Conditions {
+			if condition.Type == xpv1.TypeReady {
+				readyStatus = condition.Status
+			} else if condition.Type == xpv1.TypeSynced {
+				syncedStatus = condition.Status
+			}
+		}
+
+		if syncedStatus == corev1.ConditionTrue && readyStatus == corev1.ConditionTrue {
+			dnsStatus = corev1.ConditionTrue
+		} else if syncedStatus == corev1.ConditionUnknown || readyStatus == corev1.ConditionUnknown {
+			dnsStatus = corev1.ConditionUnknown
+		}
+
+		records = append(records, types.DNS{Status: dnsStatus, Name: (*record.Status.AtProvider.ID)[:len(*record.Status.AtProvider.ID)-1]})
+	}
+	return records
+}
+
+// prepareDNSListOptions prepares a list options for querying.
+func prepareDNSListOptions(namespace, name string) *client.ListOptions {
```

Comment:
Don't we already have similar functions? If not, maybe create one which gets the `labelSet` as a parameter from the caller which can then be re-used?